python/bindings.cpp

```diff
@@ -0,0 +1,167 @@
+        .def("KeyPoint", static_cast<void (las::Point::*)(const bool &)>(&las::Point::KeyPoint))
+        .def("Withheld", static_cast<bool (las::Point::*)() const>(&las::Point::Withheld))
+        .def("Withheld", static_cast<void (las::Point::*)(const bool &)>(&las::Point::Withheld))
+        .def("Overlap", static_cast<bool (las::Point::*)() const>(&las::Point::Overlap))
+        .def("Overlap", static_cast<void (las::Point::*)(const bool &)>(&las::Point::Overlap))
+        .def("ScannerChannel", static_cast<uint8_t (las::Point::*)() const>(&las::Point::ScannerChannel))
+        .def("ScannerChannel", static_cast<void (las::Point::*)(const uint8_t &)>(&las::Point::ScannerChannel))
+        .def("ScanDirectionFlag", static_cast<bool (las::Point::*)() const>(&las::Point::ScanDirectionFlag))
+        .def("ScanDirectionFlag", static_cast<void (las::Point::*)(const bool &)>(&las::Point::ScanDirectionFlag))
+        .def("EdgeOfFlightLineFlag", static_cast<bool (las::Point::*)() const>(&las::Point::EdgeOfFlightLineFlag))
+        .def("EdgeOfFlightLineFlag", static_cast<void (las::Point::*)(const bool &)>(&las::Point::EdgeOfFlightLineFlag))
+        .def("ClassificationBitFields",
+             static_cast<uint8_t (las::Point::*)() const>(&las::Point::ClassificationBitFields))
+        .def("ClassificationBitFields",
+             static_cast<void (las::Point::*)(const uint8_t &)>(&las::Point::ClassificationBitFields))
+        .def("Classification", static_cast<uint8_t (las::Point::*)() const>(&las::Point::Classification))
+        .def("Classification", static_cast<void (las::Point::*)(const uint8_t &)>(&las::Point::Classification))
+        .def("ScanAngleRank", static_cast<int8_t (las::Point::*)() const>(&las::Point::ScanAngleRank))
+        .def("ScanAngleRank", static_cast<void (las::Point::*)(const int8_t &)>(&las::Point::ScanAngleRank))
+        .def("ExtendedScanAngle", static_cast<int16_t (las::Point::*)() const>(&las::Point::ExtendedScanAngle))
+        .def("ExtendedScanAngle", static_cast<void (las::Point::*)(const int16_t &)>(&las::Point::ExtendedScanAngle))
+        .def("ScanAngle", static_cast<float (las::Point::*)() const>(&las::Point::ScanAngle))
+        .def("ScanAngle", static_cast<void (las::Point::*)(const float &)>(&las::Point::ScanAngle))
+        .def("UserData", static_cast<uint8_t (las::Point::*)() const>(&las::Point::UserData))
+        .def("UserData", static_cast<void (las::Point::*)(const uint8_t &)>(&las::Point::UserData))
+        .def("PointSourceID", static_cast<uint16_t (las::Point::*)() const>(&las::Point::PointSourceID))
+        .def("PointSourceID", static_cast<void (las::Point::*)(const uint16_t &)>(&las::Point::PointSourceID))
+        .def("RGB",
+             static_cast<void (las::Point::*)(const uint16_t &, const uint16_t &, const uint16_t &)>(&las::Point::RGB))
+        .def("Red", static_cast<uint16_t (las::Point::*)() const>(&las::Point::Red))
```

Comment:
remember to update these to use properties ( https://pybind11.readthedocs.io/en/stable/classes.html#instance-and-static-fields , see the part where they talk about `def_property`)